### PR TITLE
RUBY-1860 Standardize on :write_concern for write concern options

### PR DESCRIPTION
**Pull request by p-mongo**
https://jira.mongodb.com/browse/RUBY-1860

### CODE REVIEW

File: docs/tutorials/ruby-driver-crud-operations.txt [hunk 1]
```diff
@@ -482,6 +470,173 @@ Deleting
...
+When ``#with`` methods are used to alter the options on a client or collection,
+the last provided option wins in case of naming differences:
+
+.. code-block:: ruby
+
+  client = Mongo::Client.new([ '127.0.0.1:27741' ], database: 'music',
+    write_concern: {w: 2})
+  alt_client = client.with(write: {w: 3})
+  alt_client.options[:write]
+  alt_client.options[:write_concern]
```
Comment (HanaPearlman) [inline docs/tutorials/ruby-driver-crud-operations.txt]: Should this include what the resulting values are for `alt_client.options[:write]` and `alt_client.options[:write_concern]`?

Comment (p-mongo) [reply docs/tutorials/ruby-driver-crud-operations.txt]: Yes indeed, thank you.

File: lib/mongo/collection/view/builder/aggregation.rb [hunk 2]
```diff
@@ -93,7 +93,10 @@ def write?
           def aggregation_command
             command = BSON::Document.new(:aggregate => collection.name, :pipeline => pipeline)
             command[:cursor] = cursor if cursor
-            command[:readConcern] = collection.read_concern if collection.read_concern
+            if collection.read_concern
+              command[:readConcern] = Options::Mapper.transform_values_to_strings(
```
Comment (saghm) [inline lib/mongo/collection/view/builder/aggregation.rb]: All of these read concern changes don't seem related to the ticket at all; are they required for correctness, or is it possible to just make the write concern changes and make these read concern changes in a separate commit?

File: lib/mongo/client.rb [hunk 3]
```diff
@@ -352,34 +355,58 @@ def initialize(addresses_or_uri, options = nil)
...
         addresses = addresses_or_uri
       end
+
       # Special handling for sdam_proc as it is only used during client
       # construction
       sdam_proc = options.delete(:sdam_proc)
-      @options = validate_options!(Database::DEFAULT_OPTIONS.merge(options)).freeze
+
+      @options = validate_new_options!(Database::DEFAULT_OPTIONS.merge(options))
+=begin WriteConcern object support
```
[the remark this answers is not in the document]
Comment (p-mongo) [reply lib/mongo/client.rb]: It is a partial implementation of support for passing write concern objects where write concern is accepted as an option. As it is incomplete, you can disregard it for the purposes of this PR.

File: lib/mongo/collection/view/builder/aggregation.rb (hunk 2 above)
Comment (p-mongo) [reply lib/mongo/collection/view/builder/aggregation.rb]: I removed read concern changes from this PR.

File: lib/mongo/client.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (saghm) [reply lib/mongo/client.rb]: Understood
